orca-keel/src/main/kotlin/com/netflix/spinnaker/orca/keel/task/PublishKeelSpecsTask.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.keel.task
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.convertValue
+import com.netflix.spinnaker.orca.ExecutionStatus
+import com.netflix.spinnaker.orca.KeelService
+import com.netflix.spinnaker.orca.RetryableTask
+import com.netflix.spinnaker.orca.TaskResult
+import com.netflix.spinnaker.orca.igor.ScmService
+import com.netflix.spinnaker.orca.keel.model.ResourceSpec
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import retrofit.client.Response
+import java.util.concurrent.TimeUnit
+
+@Component
+class PublishKeelSpecsTask
+@Autowired
+constructor(
+  private val keelService: KeelService,
+  private val keelObjectMapper: ObjectMapper,
+  private val scmService: ScmService
+) : RetryableTask {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  override fun execute(stage: Stage): TaskResult {
+//      parameters:
```

Comment:
These are just here temporarily as a reference for what we currently have in the pre-configured job stage. If it's not obvious, I've favored convention in terms of where to look for the manifests, and have also replaced the repo URL with discrete parameters for the SCM system type, project key and repo, which seems to be the pattern used elsewhere.

The `ref` parameter is intended to contain a git reference, which would allow to specify any branch or commit, but as noted below I believe that information should come from the trigger, not the user.

Reply:
I think that you're right, the commit should come from the trigger. But, if there is no trigger, we should default to master/the latest commit. The hash can be supplied by the normal git trigger (not the rocket trigger, which is internal to netflix). In this case it makes sense to consciously support both at first!